src/main/java/com/github/osisoft/sdsjava/Program.java

```diff
@@ -452,7 +460,68 @@ public static boolean toRun() {
                 System.out.println();
             }
 
-            // Step 18
+            // Community Steps
+            if (communityId != null && communityId.length() > 0) {
+                // Step 18
+                // getting tenant roles
+                System.out.println();
+                System.out.println("Get tenant roles");
+                ArrayList<Role> roles = rolesClient.getRoles(tenantId);
+                Role role = null;
+                for (Role r : roles) {
+                    if (r.getRoleTypeId() != null && r.getRoleTypeId().equals(rolesClient.CommunityMemberRoleTypeId)
+                        && r.getCommunityId() != null && r.getCommunityId().equals(communityId)) {
+                        role = r;
+                        break;
+                    }
+                }
+                System.out.println("Community member Id:");
+                System.out.println(role.getId());
+
+                // sharing stream to community
+                System.out.println();
+                System.out.println("Sharing stream to community");
+
+                JsonArray patch = new JsonArray();
```

Comment:
would this collection of lines of code be better as a helper function?  to make it clear what is going on here?

Reply:
By that you mean the steps to build out the `patch` object? Could make sense to do that, it's especially cumbersome in Java.